iceoryx2/src/service/static_config/message_type_details.rs

```diff
@@ -266,4 +266,130 @@ mod tests {
+        let expected = 32;
+        assert_that!(sut.size(), eq expected);
+
+        let details = MessageTypeDetails::from::<i64, i64, i64>(TypeVariant::FixedSize);
+        let sut = details.sample_layout(2);
+        #[cfg(target_pointer_width = "32")]
+        let expected = 40;
+        #[cfg(target_pointer_width = "64")]
+        let expected = 48;
+        assert_that!(sut.size(), eq expected);
+
+        let details = MessageTypeDetails::from::<i64, i64, bool>(TypeVariant::FixedSize);
+        let sut = details.sample_layout(3);
+        #[cfg(target_pointer_width = "32")]
+        let expected = 24;
+        #[cfg(target_pointer_width = "64")]
+        let expected = 32;
+        assert_that!(sut.size(), eq expected);
+
+        let details = MessageTypeDetails::from::<i64, i32, bool>(TypeVariant::FixedSize);
+        let sut = details.sample_layout(11);
+
+        #[cfg(target_pointer_width = "32")]
+        let expected = 28;
+        #[cfg(target_pointer_width = "64")]
+        let expected = 32;
+        assert_that!(sut.size(), eq expected);
+
+        #[repr(C)]
+        struct Demo {
+            _b: bool,
+            _i16: i16,
+            _i64: i64,
+        }
+
+        let details = MessageTypeDetails::from::<i64, i64, Demo>(TypeVariant::FixedSize);
+        let sut = details.sample_layout(2);
+        #[cfg(target_pointer_width = "32")]
+        let expected = 48;
+        #[cfg(target_pointer_width = "64")]
+        let expected = 64;
+        assert_that!(sut.size(), eq expected);
+    }
+
+    #[test]
+    fn test_is_compatible_to() {
+        let left = MessageTypeDetails::from::<i64, i64, i8>(TypeVariant::FixedSize);
+        let right = MessageTypeDetails::from::<i64, i64, u8>(TypeVariant::FixedSize);
+        let sut = left.is_compatible_to(&right);
+        assert_that!(sut, eq false);
+
+        let left = MessageTypeDetails::from::<i64, i64, i64>(TypeVariant::FixedSize);
+        let right = MessageTypeDetails::from::<i64, i64, i32>(TypeVariant::FixedSize);
+        let sut = left.is_compatible_to(&right);
+        assert_that!(sut, eq false);
+
+        // right may have a different alignment from left.
+        // but note that the header alignment must be the same
+        let right = MessageTypeDetails {
+            header: TypeDetail {
+                variant: TypeVariant::FixedSize,
+                type_name: "i64".to_string(),
+                size: 8,
+                alignment: ALIGNMENT,
+            },
+            user_header: TypeDetail {
+                variant: TypeVariant::FixedSize,
+                type_name: "i64".to_string(),
+                size: 8,
+                alignment: 2 * ALIGNMENT,
+            },
+            payload: TypeDetail {
+                variant: TypeVariant::FixedSize,
+                type_name: "i64".to_string(),
+                size: 8,
+                alignment: 2 * ALIGNMENT,
+            },
+        };
+        // smaller to bigger is allowed.
```

Comment:
Could you create separate tests out of it with an expressive name like your comment here. We try that every test verifies just one aspect in isolation - when it makes sense.

Reply:
done in fa7e803f96655d146988a67f0a70215ce0f95880